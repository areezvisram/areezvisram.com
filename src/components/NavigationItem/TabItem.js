import React from 'react';
import { Tab, makeStyles } from '@material-ui/core';

const useStyles = makeStyles((theme) => ({
    tab: {
        color: theme.palette.primary.contrastText,        
        fontSize: theme.spacing(2.5),        
        transition: 'transform .2s',
        transitionTimingFunction: 'ease-out',
        "&:hover": {
            transform: 'scale(1.2)'
        }, 
        padding: theme.spacing(4)    
    }
}));


const TabItem = ({ label, route, component }) => {
    const classes = useStyles();
    return (        
        <Tab key={label} label={label} className={classes.tab} component={component} to={route}/>
    )
};

export default TabItem;