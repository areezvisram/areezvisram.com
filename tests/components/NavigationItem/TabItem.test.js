import React from 'react';
import { mount, shallow } from 'enzyme';
import TabItem from '../../../src/components/NavigationItem/TabItem';
import { Typography, Tab } from '@material-ui/core';

describe('<TabItem />', () => {
    let props;
    beforeEach(() => {
        props = {
            label: 'TEST',
            route: '/test',
            component: Typography
        }
    });

    it('should render as expected', () => {
        const wrapper = shallow(<TabItem {...props} />);
        expect(wrapper).toMatchSnapshot();
    });

    it('should render a Tab', () => {
        const wrapper = mount(<TabItem {...props} />);
        expect(wrapper.find(Tab).exists()).toBeTruthy();
    });

    it('should pass the props to the Tab', () => {
        const tab = mount(<TabItem {...props} />).find(Tab);
        expect(tab.prop('label')).toEqual(props.label);
        expect(tab.key()).toEqual(props.label);
        expect(tab.prop('to')).toEqual(props.route);
        expect(tab.prop('component')).toEqual(props.component);
    })
})