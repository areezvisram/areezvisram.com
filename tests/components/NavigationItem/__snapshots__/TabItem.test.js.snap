// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TabItem /> should render as expected 1`] = `ShallowWrapper {}`;
